README.md

```diff
@@ -12,8 +12,15 @@ For performance benchmarks of these hash functions and their comparison to other
 ## Merkle
 [Merkle module](./src/merkle/) provides a set of data structures related to Merkle trees. All these data structures are implemented using the RPO hash function described above. The data structures are:
 
-* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `MerkleError`: a wrapper for the possible error variants when performing Merkle tree operations, including storage failure.
+* `MerklePath`: a Merkle path container, composed of a sequence of nodes. This wrapper struct will encapsulate the Merkle opening verification.
 * `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.
+* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `NodeIndex`: a Merkle tree address for its nodes, consisting of a tuple containing `depth` and `value`.
+* `SimpleSmt`: a Sparse Merkle Tree, mapping 63-bit keys to 4-element leaf values.
```

Comment:
I would probably either remove `MerkleError`, `NodeIndex`, and `MerklePath` or maybe just mention them in a single sentence under this list. These are supporting structs which are used by `MerkleTree`, `SimpleSmt`, and `MerklePathSet`, and I don't think we need to call as much attention to them.

Reply:
Done!

Reply:
Thank you! Sorry for another request, but could we change this to something like this:

* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
* `SimpleSmt`: a Sparse Merkle Tree, mapping 63-bit keys to 4-element leaf values.
* `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.

The module also contains additional supporting components such as `NodeIndex`, `MerklePath`,  and `MerkleError`  to assist with tree indexation, opening proofs, and reporting inconsistent arguments/state.